zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceWithSuspensionInstructionTest.java

```diff
@@ -0,0 +1,560 @@
+                .endEvent()
+                .moveToActivity(elementToSuspend)
+                .endEvent()
+                .done())
+        .deploy();
+
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(processId)
+            .withRuntimeSuspendInstruction(elementToSuspend)
+            .create();
+
+    // when
+    ENGINE
+        .message()
+        .withName("myMessage")
+        .withCorrelationKey("myKey")
+        .withTimeToLive(1000)
+        .publish();
+
+    // then
+    final var result =
+        RecordingExporter.processInstanceRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .onlyEvents()
+            .limit(elementToSuspend, ProcessInstanceIntent.ELEMENT_TERMINATED)
+            .filter(
+                record ->
+                    record.getValue().getElementId().equals(processId)
+                        || record.getValue().getElementId().equals(elementToSuspend));
+
+    assertThat(result)
+        .extracting(Record::getIntent, record -> record.getValue().getElementId())
+        .containsSequence(
+            Tuple.tuple(ProcessInstanceIntent.ELEMENT_TERMINATING, elementToSuspend),
+            Tuple.tuple(ProcessInstanceIntent.ELEMENT_SUSPENDED, processId),
+            Tuple.tuple(ProcessInstanceIntent.ELEMENT_TERMINATED, elementToSuspend));
+  }
+
+  @Test
+  public void shouldNotSuspendWhenNonInterruptingBoundaryEventIsActivated() {
+    // given
+    final String processId = "process";
+    final String elementToSuspend = "element";
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(processId)
+                .startEvent()
+                .serviceTask(elementToSuspend, t -> t.zeebeJobType("jobType"))
+                .boundaryEvent("boundary")
+                .cancelActivity(false)
+                .message(
+                    messageBuilder ->
+                        messageBuilder.name("myMessage").zeebeCorrelationKey("=\"myKey\""))
+                .endEvent()
+                .moveToActivity(elementToSuspend)
+                .endEvent()
+                .done())
+        .deploy();
+
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(processId)
+            .withRuntimeSuspendInstruction(elementToSuspend)
+            .create();
+
+    // when
+    ENGINE
+        .message()
+        .withName("myMessage")
+        .withCorrelationKey("myKey")
+        .withTimeToLive(1000)
+        .publish();
+
+    // then
+    final var result =
+        RecordingExporter.processInstanceRecords()
+            .onlyEvents()
+            .limit("boundary", ProcessInstanceIntent.ELEMENT_COMPLETED)
+            .withProcessInstanceKey(processInstanceKey)
+            .limit(processId, ProcessInstanceIntent.ELEMENT_SUSPENDED)
+            .filter(record -> record.getValue().getElementId().equals(processId));
```

Comment:
🔧 It's important to filter on the process instance key before limiting to avoid encountering records from other test runs. Additionally, the secondary limit would never hit and is thus redundant.

```suggestion
    final var result =
        RecordingExporter.processInstanceRecords()
            .onlyEvents()
            .withProcessInstanceKey(processInstanceKey)
            .limit("boundary", ProcessInstanceIntent.ELEMENT_COMPLETED)
            .filter(record -> record.getValue().getElementId().equals(processId));
```